flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -317,6 +319,10 @@ public AbstractYarnClusterDescriptor createDescriptor(String defaultApplicationN
 		if (cmd.hasOption(SLOTS.getOpt())) {
 			int slots = Integer.valueOf(cmd.getOptionValue(SLOTS.getOpt()));
 			yarnClusterDescriptor.setTaskManagerSlots(slots);
+		} else if (config.containsKey(ConfigConstants.YARN_VCORES)) {
```

Comment:
I think the preference order should be `TASK_MANAGER_NUM_TASK_SLOTS` > `YARN_VCORES` for the slots.